### ISSUE TITLE
Add at-rule-semicolon-space-before rule

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2346]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

New rule `at-rule-semicolon-space-before`.

> Which rule, if any, is this issue related to?

`at-rule-semicolon-space-before`

> What CSS is needed to reproduce this issue?

```css
@import "styles/foo";
@import "styles/bar" ;
@import "styles/foobar"
;
```

> What stylelint configuration is needed to reproduce this issue?

No matter

> Which version of stylelint are you using?

No matter

> How are you running stylelint: CLI, PostCSS plugin, Node API?

No matter

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No matter

> What did you expect to happen?

No matter

> What actually happened (e.g. what warnings or errors you are getting)?

No matter

**Comment by jeddy3** [comment]
SGTM. 

As an aside, I was somewhat surprised to see this as I thought we'd pretty much covered all constructs and their punctations. There definitely can't be many left.

- Name: `at-rule-semicolon-space-before`
- Description: "Require a single space or disallow whitespace before the semicolons of at-rules."
- Primary options: `"always"|"never"|"always-single-line"|"never-single-line"`
- Secondary options: none
- Messages:
   - "Expected single space before ";""
   - "Unexpected whitespace before ";""
   - "Expected single space before ";" in a single-line at-rule"
  -  "Unexpected whitespace before ";" in a single-line at-rule"
- Section: "At-rule"

FYI, use `declaration-block-semicolon-space-before` as a blueprint.

**Comment by CAYdenberg** [comment]
@jeddy3 Are the single-line options necessary here? According to [CSS Tricks](https://css-tricks.com/the-at-rules-of-css/), all @rules that are basically either key-value pairs, or open up their own blocks, in which case the block would not have a ; following it? SCSS seems to follow this convention (see @mixin). I guess I'm having trouble thinking of a multi-line example.

**Comment by jeddy3** [comment]
The use case I was thinking of was an `@import` [followed by a `<media-query-list>`](https://developer.mozilla.org/en-US/docs/Web/CSS/@import) e.g.

```css
@import 
  url('landscape.css') 
  screen and (orientation:landscape),
  projection
;

a {}
```

We could always just start with the `"always"|"never"` options and add the `"always-single-line"|"never-single-line"` ones if there's ever a request for it.

**Comment by CAYdenberg** [comment]
For anyone interested, I started work on this here: https://github.com/CAYdenberg/stylelint/tree/issue_2346

**Comment by evilebottnawi** [comment]
@CAYdenberg can you do PR?

**Comment by CAYdenberg** [comment]
Sure ... I wasn't sure if I could tag it. It's a WIP.

**Comment by evilebottnawi** [comment]
@CAYdenberg that's okay, we will help to complete it

**Comment by hudochenkov** [comment]
Done in https://github.com/stylelint/stylelint/pull/2388

Thank you for your work, @CAYdenberg!
